### PR TITLE
yash: 2.54 -> 2.55

### PR DESCRIPTION
**Pull request by qbit**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Result of `nixpkgs-review pr 251361` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>getoptions</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>yash</li>
  </ul>
</details>

**Comment by matthiasbeyer** [comment]
`getoptions` does not build anymore with this.
I think that `getoptions` does not build anymore because it tests itself against `yash` and that test broke by the update.

I am not sure how to proceed, `getoptions` maintainer @matrss should decide this!

**Comment by matrss** [comment]
The test suite for getoptions against yash has been broken since the last update of yash already. I wrote something about that here: https://github.com/NixOS/nixpkgs/pull/218264#pullrequestreview-1434402054. I filed an issue with shellspec, since I think the issue is coming from there, but nothing was done about it yet. I'd say we should just disable the tests of getoptions against yash for now.

File: pkgs/shells/yash/default.nix
Comment (matrss) [inline pkgs/shells/yash/default.nix:10]: Can this become a `fetchFromGitHub` since we get the source from github.com?

**Review by matrss — CHANGES_REQUESTED** [review]

File: pkgs/shells/yash/default.nix
Comment (qbit) [reply pkgs/shells/yash/default.nix:10]: Tried switching to it, but for some reason it can't find `xgettext`:

```
xgettext -d yash -D .. -cTRANSLATORS: -f - -F --no-wrap \
        --package-name=yash --package-version=2.55 \
        --msgid-bugs-address=http://osdn.jp/projects/yash/forums/ \
        --copyright-holder='Magicant' -kgt -kNgt -kngt:1,2 --flag=sb_vprintf:1:c-format --flag=sb_printf:1:c-format --flag=malloc_vprintf:1:c-format --flag=malloc_printf:1:c-format --flag=xerror:1:c-format --flag=serror:1:c-format --flag=le_compdebug:1:c-format
/nix/store/9b4skch97sv02rwmczbxp1pwg382hv84-bash-5.2-p15/bin/bash: line 2: xgettext: command not found
```

Comment (qbit) [reply pkgs/shells/yash/default.nix:10]: ah, got it - was missing asciidoc and gettext as a nativebuildinput

**Review by matrss — APPROVED** [review]
LGTM. Build works (x86_64-linux) and the binary starts fine.

**Pull request merged**
